osu.Game/Screens/SelectV2/BeatmapCollectionStore.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Game.Collections;
+using osu.Game.Database;
+using Realms;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public partial class BeatmapCollectionStore : Component
+    {
+        [Resolved]
+        private RealmAccess realm { get; set; } = null!;
+
+        private IDisposable? realmSubscription;
+
+        private readonly BindableList<Live<BeatmapCollection>> liveCollections = new BindableList<Live<BeatmapCollection>>();
+        private readonly List<BeatmapCollection> detachedCollections = new List<BeatmapCollection>();
+
+        /// <summary>
+        /// Gets a thread-safe bound copy to the list of <see cref="BeatmapCollection"/> present in the user's database, wrapped in <see cref="Live{T}"/> objects.
```

Comment:
the "thread safe" part of this comment is really confusing, because a reader will not know what it implies.

thread safe usually refers to a class which enforces thread safety, not an arbitrary type. i don't think this comment should exist at all.